src/queries/a11yState.ts

```diff
@@ -0,0 +1,68 @@
+import type { ReactTestInstance } from 'react-test-renderer';
+import { AccessibilityState } from 'react-native';
+import { makeQueries } from './makeQueries';
+import type {
+  FindAllByQuery,
+  FindByQuery,
+  GetAllByQuery,
+  GetByQuery,
+  QueryAllByQuery,
+  QueryByQuery,
+} from './makeQueries';
+
+export function matchObject<T extends Record<string, unknown>>(
+  prop: T | undefined,
+  matcher: T
+): boolean {
+  return prop
+    ? Object.keys(matcher).length !== 0 &&
+        Object.keys(prop).length !== 0 &&
+        !Object.keys(matcher).some((key) => prop[key] !== matcher[key])
+    : false;
+}
+
+const queryAllByA11yState = (
+  instance: ReactTestInstance
+): ((accessibilityStateKey: AccessibilityState) => Array<ReactTestInstance>) =>
+  function queryAllByDisplayValueFn(accessibilityStateKey) {
+    return instance.findAll(
+      (node) =>
+        typeof node.type === 'string' &&
+        matchObject(node.props.accessibilityState, accessibilityStateKey)
+    );
+  };
+
+const getMultipleError = (a11yState: AccessibilityState) =>
+  `Found multiple elements with accessibilityState: ${JSON.stringify(
+    a11yState
+  )}`;
+const getMissingError = (a11yState: AccessibilityState) =>
+  `Unable to find an element with accessibilityState: ${JSON.stringify(
+    a11yState
+  )}`;
+
+const { getBy, getAllBy, queryBy, queryAllBy, findBy, findAllBy } = makeQueries(
+  queryAllByA11yState,
+  getMissingError,
+  getMultipleError
+);
+
+export type ByA11yStateQueries = {
```

Comment:
Add alias names `getByAccessiblityState`, etc. See comments in `a11yHint.ts` for details.